### ISSUE TITLE
Hai & humans do not sign a treaty if you fail to help Nanachi or Turner

### ISSUE THREAD

**Issue by UnorderedSigh** [endless-sky/endless-sky#8257]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hai Reveal part 7 is blocked by two mission chains, which makes the Hai-human treaty a secret, rather than a major galactic event.

1. Expanding Business - only triggers on one planet
2. Nanachi - 15% chance on only one planet. You have to visit the Giverstone spaceport *a lot* to get that mission intentionally.

If you fail either of those mission chains, then the Hai and humans will never sign a treaty. To put it in perspective: it is *so important* to all Hai and humans that one billionaire gets richer, and a young Hai pilot gets her own James, that there'll be no treaty if you fail. That makes *no sense at all.*

#### Suggested Changes

I think this is a more reasonable alternative:

1. Nanachi gets 100% chance after the blockade is lifted, and is blocked after "event: hai-human treaty prepared"
2. Expanding Business: blocked after "event: hai-human treaty prepared"
3. Hai Reveal [D01A] Summoned to Mountaintop won't trigger if the Nanachi or Expanding Business stories are active.

### Steps to Reproduce

Read the txt files; it's the only way to see it.

The conditions for the Hai and humans signing a treaty (Hai Reveal part 7) includes these two mission chains:

```ruby
mission "Hai Reveal [D01A] Summoned to Mountaintop"
        ...
        to offer
                ...
                has "Expanding Business [4]: done"
                has "Nanachi 4: done"
```

If you fail to help Nanachi or Turner, the Hai and humans will never sign a treaty!

Here's Nanachi's mission, well-hidden behind its 15% chance and single planet (Giverstone)

```ruby
mission "Nanachi 1"
        source "Giverstone"
        ...
        to offer
                random < 15
                ...
```

### Expected Behavior

The signing of the Hai-human treaty should be easy to find. It should not be secret.

### Operating System

`Linux 5.0.0-37-generic #40~18.04.1-Ubuntu SMP Thu Nov 14 12:06:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

### Game Source

Built from source

### Game Version

28780a902

**Comment by Amazinite** [comment]
No longer relevant to the current state of HR.
